### ISSUE TITLE
`use_key_in_widget_constructors` is not handling `super_parameters`

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3323]
For the below code with SDK `main`

```dart
import 'package:flutter/material.dart';

class MyWidget extends StatelessWidget {
  const MyWidget({super.key});

  @override
  Widget build(BuildContext context) {
    throw UnimplementedError();
  }
}

```

**Expected behavior**
No lint

**Actual behavior**
[Lint](https://dart-lang.github.io/linter/lints/use_key_in_widget_constructors.html) at the `const` saying: `Use key in widget constructors`

**Comment by pq** [comment]
This *should* be fixed w/ https://github.com/dart-lang/linter/pull/3295 and be in the latest `dev` build.  What SDK are you using?

**Comment by asashour** [comment]
I am on the `main` SDK (`2.1.7.0-edge`), with probably misconfiguration somewhere.

Thanks for your input, and sorry for bothering.
